pkg/ec2provider/ec2provider_test.go

```diff
@@ -0,0 +1,148 @@
+package ec2provider
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	DescribeDelay = 100
+)
+
+type mockEc2Client struct {
+	ec2iface.EC2API
+	Reservations []*ec2.Reservation
+}
+
+func (c *mockEc2Client) DescribeInstances(in *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	// simulate the time it takes for aws to return
+	time.Sleep(DescribeDelay * time.Millisecond)
+	var reservations []*ec2.Reservation
+	for _, res := range c.Reservations {
+		var reservation ec2.Reservation
+		for _, inst := range res.Instances {
+			for _, id := range in.InstanceIds {
+				if aws.StringValue(id) == aws.StringValue(inst.InstanceId) {
+					reservation.Instances = append(reservation.Instances, inst)
+				}
+			}
+		}
+		if len(reservation.Instances) > 0 {
+			reservations = append(reservations, &reservation)
+		}
+	}
+	return &ec2.DescribeInstancesOutput{
+		Reservations: reservations,
+	}, nil
+}
+
+func setup() *ec2ProviderImpl {
```

Comment:
Can you name this something more specific like `newMockedEC2ProviderImpl`?

Reply:
Done , good name suggestions